src/formatDate.js

```diff
@@ -8,7 +8,85 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [aFrom, bFrom, cFrom, separator] = fromFormat;
+  const fromFormatArray = [aFrom, bFrom, cFrom];
+  const [aTo, bTo, cTo, newSeparator] = toFormat;
+  const toFormatArray = [aTo, bTo, cTo];
+  const dateinNumbers = date.split(separator);
```

Comment:
Ensure that the `date` string contains the expected number of segments when split by `separator`. If the `date` string does not match the expected format, this could lead to unexpected behavior or errors.